tests/spread/smoketests/basic-centos-7/task.yaml

```diff
@@ -1,38 +1,34 @@
 summary: pack a simple init-created charm for CentOS 7
+# This test should be nearly identical to the basic test, but only runs on 18.04
 systems:
   # CentOS 7 only works in 18.04 because it needs cgroup v1
   - ubuntu-18.04-64
 kill-timeout: 30m
 
-prepare: |
-  charmcraft init --project-dir=charm
-  pushd charm
+environment:
+  BASE_NAME: centos
+  BASE_CHANNEL: "7"
 
-  cat <<- EOF >> charmcraft.yaml
-  name: basic-centos-7
-  type: charm
-  summary: A basic charm for CentOS 7
-  description: |
-    This is a basic charm for CentOS 7.
+include:
+  - tests/
 
-  bases:
-    - build-on:
-      - name: centos
-        channel: "7"
-      run-on:
-      - name: centos
-        channel: "7"
-  EOF
+prepare: |
+  tests.pkgs install unzip
+  charmcraft init --project-dir=charm
 
-  popd
+  sed -i "s/- name: ubuntu/- name: ${BASE_NAME}/g" charm/charmcraft.yaml
+  sed -i "s/channel:.*/channel: '${BASE_CHANNEL}'/g" charm/charmcraft.yaml
```

Comment:
Is it safe to replace inline in this case, instead of using a template file as source?

Reply:
It is safe, if the channel misplaced, pydantic should given an error.